Sources/SwiftProtobuf/TextFormatEncodingVisitor.swift

```diff
@@ -372,18 +401,31 @@ internal struct TextFormatEncodingVisitor: Visitor {
   // Messages and groups
   mutating func visitRepeatedMessageField<M: Message>(value: [M],
                                              fieldNumber: Int) throws {
+      // Look up field name against outer message encoding state
+      let fieldName = formatFieldName(lookingUp: fieldNumber)
+      // Cache old encoder state
+      let oldNameMap = self.nameMap
+      let oldNameResolver = self.nameResolver
+      let oldExtensions = self.extensions
+      // Update encoding state for new message type
+      self.nameMap = (M.self as? _ProtoNameProviding.Type)?._protobuf_nameMap
+      self.nameResolver = [:]
+      self.extensions = (value as? ExtensibleMessage)?._protobuf_extensionFieldValues
+      // Iterate and encode each message
       for v in value {
-          emitFieldName(lookingUp: fieldNumber)
+          encoder.emitFieldName(name: fieldName)
```

Comment:
Seems like all repeated handlings could use the speedup of calcing the name once?

Reply:
I went ahead and did this for all the repeated handlings, though I haven't actually been able to measure any difference.

The real reason I need it for repeated message fields is not really for performance but because of the context switch; the field name has to be looked up against the outer message.  Otherwise, I would have to repeat the context switch for every entry.  (Though honestly, that might not be too bad performance-wise; the context switch is pretty light.)